### PR TITLE
Ternary operator

### PR DESCRIPTION
**Pull request by lushoBarlett**
Implements the ternary operator.

~Currently it breaks in the typechecker...~ The Address Sanitizer complains about an illegal access in `typecheck(CallExpression)`, but I can't seem to understand why. I checked if it was a parsing mistake where it thought it was a CallExpression instead of something else, but I figured this should not be the case. I need help to understand if I misused the type system or something.

### CODE REVIEW

File: src/typecheck.cpp [hunk 1]
```diff
@@ -200,6 +200,18 @@ void typecheck(TypedAST::IndexExpression* ast, Frontend::CompileTimeEnvironment&
...
+	typecheck(ast->m_condition.get(), env);
+	env.m_typechecker.m_core.unify(
+	    ast->m_condition->m_value_type, env.m_typechecker.mono_boolean());
+
+	typecheck(ast->m_then_expr.get(), env);
+	typecheck(ast->m_else_expr.get(), env);
+
+	env.m_typechecker.m_core.unify(
+	    ast->m_then_expr->m_value_type, ast->m_else_expr->m_value_type);
+}
```
Comment (SebastianMestre) [inline src/typecheck.cpp:213]: `ast->m_value_type` is never assigned

**Review by SebastianMestre — CHANGES_REQUESTED** [review]

**Comment by lushoBarlett** [comment]
Okay so, I was a little confused as to _why_ something like `i == 0` even worked... After seeing how the `TypedAST` completely ignored it. Then I remembered that every bin-op is translated into a native function, so then it hit me why the failure ocurrs in a `CallExpression`. It's the `+` *function* that's getting called.

I probably need to check out where in the codebase that transformation happens, because I don't remember.

**Comment by SebastianMestre** [comment]
The code looks good to me, and the tests pass... I'm not quite clear on whether it works, though. Your PR description seems to imply it doesn't. Does it work?

**Comment by SebastianMestre** [comment]
Ok that's all cleared up, then

**Review by SebastianMestre — APPROVED** [review]
I have no comments, good job

**Comment by lushoBarlett** [comment]
Just to leave record, the code **works**. I just left the problem (and the discovery of the problem) as a reminder of what happened. Also I felt like dashing all of that text would look bad.

**Pull request merged**